src/python/modules/ensembl/compara/citest/tabletest.py

```diff
@@ -0,0 +1,398 @@
+
+    @staticmethod
+    def _get_pipeline_table(obj_connection_engine):
+        """Get the list of tables in the database
+        Args:
+            obj_connection_engine: a connection engine (complex from sqlalchemy)
+        Returns:
+            list of table name (list of string)
+        """
+        tabl_name = 0
+        lst_pipeline_table_names = []
+        str_request = "SHOW TABLE STATUS WHERE Rows > 0;"
+        obj_request = text(str_request)
+        obj_connection = obj_connection_engine.connect()
+        obj_result = obj_connection.execute(obj_request)
+        for row in obj_result:
+            str_table_name = row[tabl_name]
+            lst_pipeline_table_names.append(str_table_name)
+        obj_connection.close()
+        return lst_pipeline_table_names
+
+    def run_tests(self):
+        """Main function running the table tests
+
+        For each tables intialised in initialise_tests this method will
+        compare the number of row between the reference and target db
+        and if some table has group by test and/or column test it will carry on
+        these tests.
+            Args:
+                None
+            Returns:
+                None
+        """
+        for dic_tabl_test in self.table_tests:
+            str_table_name = dic_tabl_test["name"]
+            int_line_variation = dic_tabl_test["variation"]
+            lst_group_by = dic_tabl_test["lst_group_by"]
+            lst_columns = dic_tabl_test["columns"]
+
+            #First it compare the number of row between the ref and target db
+            self._test_table_number_row(str_table_name, int_line_variation)
+
+            #test the group if defined in json
+            if not lst_group_by == []:
+                self._test_table_number_row_by_group(str_table_name, lst_group_by)
+
+            #test field if present in json
+            if not lst_columns == []:
+                self._test_table_by_column(str_table_name, lst_columns)
+
+    def _test_table_number_row(self, str_table_name, flt_variation):
+        """Function comparing the number of row between a table in the ref and target DB
+
+        For each tables intialised in initialise_tests this method will
+        compare the number of row between the reference and target db
+        and if some table has group by test and/or column test it will carry on
+        these tests. The result of the test is store as dictionary
+        in self.test_results attribute
+            Args:
+                str_table_name: the name of the table to test (str)
+                flt_variation: the variation (flt)
+            Returns:
+                None
+        """
+        result = {}
+        #do the check for the given table
+        int_nb_row_ref = self._get_number_row(self.ref_connection_engine, str_table_name)
+        int_nb_row_tar = self._get_number_row(self.target_connection_engine, str_table_name)
```

Comment:
Not a big fan of `_tar`. I can't help thinking of a tar archive. What about `tgt` ? Not great either. Alternatively, you can use `got` and `exp` like in Perl tests